### PR TITLE
Updated documentation with API and config information

### PR DESCRIPTION
**Pull request by rchild-okta**
RESOLVES OKTA-94210, OKTA-100557, OKTA-102385
Bacon: test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,68 +1,873 @@
 [![Build Status](https://travis-ci.org/okta/okta-signin-widget.svg?branch=master)](https://travis-ci.org/okta/okta-signin-widget)
 
-<img src="/assets/img/okta-signin-widget.png" title="Okta Sign-In Widget" align="right"/>
```
Comment (ukilon-okta) [inline README.md]: no more fancy logo? :trollface:

File: README.md [hunk 2]
```diff
@@ -1,68 +1,873 @@
...
+    // The user has not been logged out, perform some error handling here.
+    return;
+  }
+  // The user is now logged out. Render the Sign-In Widget.
+});
+```
+
+## `OIDC` token.hasTokensInUrl()
+
+Synchronous method to check for access or id tokens in the url. This is used when `authParams.display = 'page'`. Returns `true` if there are tokens, and `false` if the redirect flow has not taken place yet.
```
Comment (mystiberry-okta) [inline README.md]: id tokens should be ID Tokens everywhere (we're following the spec conventions).

File: README.md [hunk 3]
```diff
@@ -1,68 +1,873 @@
...
+
+    // Use OktaSignIn
+    var signIn = new OktaSignIn(/* configOptions */);
+    ```
+
+# API
+
+## new OktaSignIn(config)
+
+Creates a new instance of the Sign-In Widget with the provided config options. The widget has many config options to customize and change its behavior, which you can read about in the [Configuration](#configuration) section. The only required option to get started is `baseUrl`, the base url for your Okta domain.
```
Comment (mystiberry-okta) [inline README.md]: To make a shorter, simpler read, we usually link to stuff without telling the reader they can read about the stuff in the link, so I'd recommend "The widget has many [config options](#configuration). The only required....

File: README.md [hunk 4]
```diff
@@ -1,68 +1,873 @@
...
+var signIn = new OktaSignIn({baseUrl: 'https://acme.okta.com'});
+```
+
+## renderEl(options, success, error)
+
+Renders the widget to the DOM, and passes control back to your app through the success and error callback functions when the user has entered a success or error state.
+
+- `options`
+  - `el` - CSS selector for the container element to attach the widget to.
+- `success` - Function that is called when the user has completed an auth flow.
```
Comment (mystiberry-okta) [inline README.md]: We usually spell out authentication.

File: README.md (hunk 2 above)
Comment (rchild-okta) [inline README.md]: Ah, good call, will update

File: README.md [hunk 5]
```diff
@@ -1,68 +1,873 @@
...
+```javascript
+var signIn = new OktaSignIn({baseUrl: 'https://acme.okta.com'});
+```
+
+## renderEl(options, success, error)
+
+Renders the widget to the DOM, and passes control back to your app through the success and error callback functions when the user has entered a success or error state.
+
+- `options`
+  - `el` - CSS selector for the container element to attach the widget to.
```
Comment (mystiberry-okta) [inline README.md]: How about "CSS selector which identifies the container element that the widget attaches to." (ending a sentence with a preposition is okay if the object of the preposition is clear, but having the first "to" makes the sentence a bit harder to read).

File: README.md [hunk 6]
```diff
@@ -1,68 +1,873 @@
...
-| **widgetOptions** | Options that are passed to the widget on initialization - reference the [developer docs](http://developer.okta.com/docs/guides/okta_sign-in_widget.html#customizing-widget-features-and-text-labels-with-javascript) to see a complete list of configurable features. |
-| **serverPort** | The port the local server runs on. Defaults to *3000* |
+| **widgetOptions** | Config options that are passed to the widget on initialization. |
+| **serverPort** | The port the local server runs on. Defaults to `3000` |
 
-## Build and test commmands
+## Build and test commands
 
 | Command | Description |
 | --- | --- |
```
Comment (mystiberry-okta) [inline README.md]: Content missing here? Also, it seems like its reference info, not part of the how-to steps.

File: README.md [hunk 7]
```diff
@@ -1,68 +1,873 @@
...
+
+- **features.smsRecovery** - Allow users with a configured mobile phone number to recover their password using an SMS message. Defaults to `false`.
+
+- **features.callRecovery** - Allow users with a configured mobile phone number to recover their password using a voice call. Defaults to `false`.
+
+- **features.windowsVerify** - Display instructions for enrolling a windows device with Okta Verify. Defaults to `false`.
+
+- **features.selfServiceUnlock** - Display the "Unlock Account" link to allow users to unlock their accounts. Defaults to `false`.
+
+- **features.multiOptionalFactorEnroll** - Allow users to enroll in multiple optional factors before finishing the auth flow. Default behavior is to force enrollment of all required factors and skip optional factors. Defaults to `false`.
```
Comment (mystiberry-okta) [inline README.md]: If you're going to use "auth flow" throughout, please use "authentication flow (auth flow)" the first time. Or spell it out everywhere.

File: README.md [hunk 8]
```diff
@@ -1,68 +1,873 @@
...
+
+    ```javascript
+    // Load the Sign-In Widget module
+    var OktaSignIn = require('@okta/okta-signin-widget');
+
+    // Use OktaSignIn
+    var signIn = new OktaSignIn(/* configOptions */);
+    ```
+
+# API
```
Comment (mystiberry-okta) [inline README.md]: I understand that CDN is an alternative to npm, but what about the instructions at the end of this file? Is that another alternative or steps you take after the CDN or npm steps? And why is the API reference (JavaScript library?) in the middle?

Comment (rchild-okta) [inline README.md]: Good question - I've been thinking of it as:
- The install section talks about how you will consume the widget
- The "developing the sign-in widget" section talks from the dev perspective (i.e. us or anyone that wants to contribute) of cloning the repo, modifying the actual source code, and building your own version.

File: README.md (hunk 7 above)
Comment (rchild-okta) [inline README.md]: Yep, have made the changes in my local branch and will push an update once I'm through this first round of comments

File: README.md (hunk 6 above)
Comment (rchild-okta) [inline README.md]: Ah, this is how github markdown works for tables (need the separator row for it to show up as a table)

**Comment by rchild-okta** [comment]
@mystiberry-okta pushed an update to address your first round of comments

File: README.md (hunk 1 above)
Comment (rchild-okta) [inline README.md]: Heh, it didn't look great on npm so I took it out

**Comment by rchild-okta** [comment]
@ukilon-okta agree - this is the first pass of "get all the information doc'ed". Later on we can think about breaking it out into multiple pages (although I am a bit partial to having all the information in one place).

**Pull request merged**